src/core/ext/filters/client_channel/lb_policy/grpclb/grpclb.c

```diff
@@ -775,8 +787,10 @@ static void create_rr_locked(grpc_exec_ctx *exec_ctx, glb_lb_policy *glb_policy,
 /* glb_policy->rr_policy may be NULL (initial handover) */
 static void rr_handover_locked(grpc_exec_ctx *exec_ctx,
                                glb_lb_policy *glb_policy) {
-  GPR_ASSERT(glb_policy->serverlist != NULL &&
-             glb_policy->serverlist->num_servers > 0);
+  GPR_ASSERT((glb_policy->fallback && glb_policy->fallback_addresses != NULL &&
```

Comment:
I agree that simpler conditions are better, at least for asserts.

But I haven't found a better way to separate the conditions when failure happens in this case. I think using two const bools to group the conditions is more readable and I will do that. But finally both bools will sit in the same assert. If the assert fails, we still can't tell which condition fails (can we?).

Do you think I should add two logs, each of which prints the failed condition?